chore(rln-relay): namespaced flags

rymnc opened the pull request:
Should resolve https://github.com/status-im/nwaku/issues/1196

This PR namespaces few flags that are exclusively used with rln-relay. This will 
unblock https://github.com/status-im/infra-role-nim-waku/pull/5 and 
https://github.com/status-im/infra-nim-waku/pull/61

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 3cbda3f6 | [#1](https://ci.status.im/job/nim-waku/job/prs/job/macos/job/PR-1204/1/) | 2022-09-30 13:09:43 | ~16 min | `macos` | [:package:`bin`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/wakunode2-220930-125321-3cbda3-macos.bin) |

jakubgs commented on examples/v2/config_chat2.nim:
```diff
@@ -270,24 +270,25 @@ type
       desc: "Address of membership contract on an Ethereum testnet", 
       defaultValue: ""
-      name: "eth-mem-contract-address" }: string
+      name: "rln-relay-eth-membership-contract-address" }: string
```
Or the `membership` part. As far as I can tell there are no other contracts, so no need to distinguish between them.

rymnc replied:
Dropping `membership` in favour of `address`

rymnc replied:
Addressed in [3f3bc1e](https://github.com/status-im/nwaku/pull/1204/commits/3f3bc1e8a7681c1ae40de402ab6ca8cd0eb26140)

jm-clius commented on examples/v2/config_chat2.nim:
```diff
@@ -270,24 +270,25 @@ type
-      name: "eth-account-address" }: string
+      name: "rln-relay-eth-account-address" }: string
     
     rlnRelayEthAccountPrivKey* {.
```
```suggestion
    rlnRelayEthAccountPrivateKey* {.
```

For consistency the metric field should match the config `name`, except that `-` (dashes) are converted to `camelCase`

rymnc replied:
Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

jm-clius commented on examples/v2/config_chat2.nim:
```diff
@@ -270,24 +270,25 @@ type
+      # NOTE: This can be derived from the private key, but kept for future use
       defaultValue: ""
-      name: "eth-account-address" }: string
+      name: "rln-relay-eth-account-address" }: string
```
```suggestion
      name: "rln-relay-eth-account" }: string
```

To match the field `rlnRelayEthAccount` (or "Address" can be suffixed to the field)

rymnc replied:
Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

rymnc replied:
Chose to continue with the address suffix to match the other config items, updated the variable

jm-clius commented on examples/v2/config_chat2.nim:
```diff
@@ -270,24 +270,25 @@ type
       desc: "Account private key for an Ethereum testnet",
       defaultValue: ""
-      name: "eth-account-privatekey" }: string
+      name: "rln-relay-eth-account-privatekey" }: string
```
```suggestion
      name: "rln-relay-eth-account-private-key" }: string
```
(Throughout most of the codebase "private key" is considered two words, though I'm sure it's up for debate :) )

rymnc replied:
Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

jm-clius commented on waku/v2/node/config.nim:
```diff
@@ -155,23 +155,24 @@ type
+      # NOTE: This can be derived from the private key, but kept for future use
       defaultValue: ""
-      name: "eth-account-address" }: string
+      name: "rln-relay-eth-account-address" }: string
```
```suggestion
      name: "rln-relay-eth-account" }: string
```

rymnc replied:
Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

jm-clius commented on waku/v2/node/config.nim:
```diff
@@ -155,23 +155,24 @@ type
-      name: "eth-account-address" }: string
+      name: "rln-relay-eth-account-address" }: string
 
     rlnRelayEthAccountPrivKey* {.
```
```suggestion
    rlnRelayEthAccountPrivateKey* {.
```

rymnc replied:
Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

jm-clius commented on waku/v2/node/config.nim:
```diff
@@ -155,23 +155,24 @@ type
       desc: "Account private key for the Ethereum testnet Goerli",
       defaultValue: ""
-      name: "eth-account-privatekey" }: string
+      name: "rln-relay-eth-account-privatekey" }: string
```
```suggestion
      name: "rln-relay-eth-account-private-key" }: string
```

rymnc replied:
Addressed in [0395e29](https://github.com/status-im/nwaku/pull/1204/commits/0395e29b0706bca9d58705bea5e6e41e6f27afaf)

jm-clius reviewed (approved):
I have added a couple of comments re consistency. In general, the `fieldName` for the config item should match the `--config-name`, but with dashes replaced with camelCase.

Once the consistency issue is addressed, no further comments from me. LGTM.

staheri14 reviewed (approved):
Thanks, looks good to me!

rymnc commented:
> > I have added a couple of comments re consistency. In general, the `fieldName` for the config item should match the `--config-name`, but with dashes replaced with camelCase.
> > Once the consistency issue is addressed, no further comments from me. LGTM.
> 
> I can still see some exceptions, but probably are fine like that @jm-clius

Addressed in [4cf046e](https://github.com/status-im/nwaku/pull/1204/commits/4cf046ebe3af84a4dcdde241bff080dfad54b2d2)
